### PR TITLE
Up to Date Coveo Zapier Integration

### PR DESCRIPTION
**Pull request by ibarnold19**
Progress on Coveo Zapier Integration. CompressedBinaryDataFileData currently being worked on as platform is giving back strange errors.

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    throw new Error(error);
+  }
+
+  throw error;
+
+};
+
+const fetchFile = (url) => {
+
+  const details = {
```
Comment (jdevost) [inline utils.js]: This shouldn't be outside the promise handler.

File: utils.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      details.size = response.headers.get('content-length');
+      const disposition = response.headers.get('content-disposition');
+
+      if (disposition) {
+        details.filename = contentDisposition.parse(disposition).parameters.filename;
+        details.contentType = path.extname(details.filename);
+      }
+
+
+      if(details.contentType === 'undefined' || details.contentType === 'null' || details.contentType === ''){
```
Comment (jdevost) [inline utils.js]: can details.contentType really be the stings `'undefined'` or `'nyll'` ?

File: utils.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        details.contentType = '.' + response.headers.get('content-type').split('/')[1].split(';')[0];
+      }
+
+      return response.buffer();
```
Comment (jdevost) [inline utils.js]: can we reduce the 2nd `.then()` to `details.content = reponse.buffer();` here

File: .zapierapprc [hunk 4]
```diff
@@ -1,4 +1,4 @@
 {
-  "id": 955,
-  "key": "App955"
+  "id": 4771,
```
[the remark this answers is not in the document]
Comment (ibarnold19) [reply .zapierapprc]: These are application specific yes but the Zapier application needs them to be pushed to Zapier whenever we update the app. Would adding .zapierapprc to .gitirgnore make this possible?

File: resources/delete.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    {key: 'sourceId', label: 'Source ID'},
+    {key: 'sourceOwner', label: 'Source Owner'},
+    {key: 'sourceType', label: 'Source Type'},
+    {key: 'numDocs', label: 'Number of Documents in Source'},
+    {key: 'title', label: 'Pushed Document Title'},
+    {key: 'numFields', label: 'Number of Fields the Source Uses'},
+    {key: 'docSize', label: 'Size (in bytes) of all the Documents in the Source'},
+  ];
+
+  for(var i = 0; i < numFields; i++){
```
Comment (jdevost) [inline resources/delete.js]: Why we need to add these `Field #i` ? 
Please add a comment here.

File: resources/push.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+};
+
+module.exports = {  
+  key: 'push',
+  noun: 'Push',
+
+  sample: {
+    docId: 'file://folder/my-file.html',
+    sourceId: 'rp5rxzbdz753uhndklv2ztkfgy-mycoveocloudv2organizationg8tp8wu3',
```
Comment (jdevost) [inline resources/push.js]: Is this sample used in the UI (shown to the user) ?
Change it to something that doesn't look like some garbage: `your-source-id` and `your-coveo-org-id`

File: test/creates/delete.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+// require('should');
```
Comment (jdevost) [inline test/creates/delete.js]: This whole file is commented.
If it isn't required anymore, delete it.

File: utils.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+  let zipDetails ={
+    content: '',
+    contentType: '',
+    size: 0,
+    filename: '',
+  };
+
+  const JSZIP = require('jszip');
+  var zip = new JSZIP();
```
Comment (jdevost) [inline utils.js]: don't use `var`

**Pull request closed without merging**